### PR TITLE
Test Webhook in class + Update README

### PR DESCRIPTION
**Pull request by ian-Liaozy**
**As a** developer,
**I need** to update the README.md file with accurate and detailed deployment instructions,
**So that** other contributors should be able to successfully deploy and verify the application.
 
### Details and Assumptions
 * The README.md will include instructions for setting up the Kubernetes cluster, building and pushing the Docker image, deploying PostgreSQL and application resources, and accessing the microservice.
 * All steps will be tested and verified before updating the file to ensure accuracy.
   
 ### Acceptance Criteria  
   
 ```
Given the deployment instructions in the updated README.md file  
When a team member follows the documented steps  
Then they should be able to successfully deploy and verify the application  
And the application should be accessible externally and connected to the PostgreSQL database.
```

This is the final version of our readme.md file. Please let me know if you have any suggestion on it. thanks!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -188,7 +158,119 @@ make run
...
+   - Check the status of Kubernetes resources:
+     ```bash
+     kubectl get all
+     ```
+   - Ensure that pods are in the `Running` state and services are available.
+
+4. **Access the Application Using Ingress**:
+   - Access the application at:
+     ```
+     http://localhost:8080
```
Comment (yc6714) [inline README.md]: Lets add more details in the steps for local kubernates deployment


Step 1: Create the Kubernetes Cluster
`make cluster
`
Step 2: Build and Prepare the Docker Image
If you haven’t already built the Docker image for the application, or if you need to update the image, perform the following steps:
Build the Docker Image Navigate to the directory containing your Dockerfile and run:
`docker build -t promotions:latest .
`

2. Check Existing Docker Images To verify the image was built successfully or check if it already exists, use:
`docker images`
3. Tag and Push the Docker Image :
```
docker tag promotions:latest cluster-registry:5000/promotions:latest
docker push cluster-registry:5000/promotions:latest
```
4. Deploy PostgreSQL and Application Resources
```
kubectl apply -f k8s/postgresql/
kubectl apply -f k8s/
```
5. Verify the Deployment
`kubectl get pods`

6. Access the application at:
     ```
     http://localhost:8080
     ```

File: README.md [hunk 2]
```diff
@@ -188,7 +158,119 @@ make run
...
+
+   ```bash
+   behave
+   ```
+
+## Kubernetes Deployment
+
+This microservice is deployed to a Kubernetes cluster using the manifests provided in the `k8s/` directory.
+
+### Local Development Setup
```
Comment (pennyyang17) [inline README.md]: Do you mean Local Deployment Setup here?

File: README.md (hunk 1 above)
Comment (ian-Liaozy) [reply README.md]: I have modify the steps in the new commit. Thanks!

File: README.md (hunk 2 above)
Comment (ian-Liaozy) [reply README.md]: Yes I think you are correct. I have modify the wording in the new commit. Thanks!

**Review by raghavm1 — APPROVED** [review]
Looks good, thanks Ian!

**Review by pennyyang17 — APPROVED** [review]
Thanks Ian!! Looks great

**Review by yc6714 — APPROVED** [review]

**Review by wencfy — APPROVED** [review]

**Pull request merged**